drivers/ethernet/eth_stm32_hal.c

```diff
@@ -28,22 +28,137 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include "eth_stm32_hal_priv.h"
 
 #if defined(CONFIG_ETH_STM32_HAL_USE_DTCM_FOR_DMA_BUFFER) && \
-	    !DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
+			!DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
 #error DTCM for DMA buffer is activated but zephyr,dtcm is not present in dts
 #endif
 
+static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef * heth,
+											   uint32_t PHYAddr,
+											   uint32_t PHYReg,
+											   uint32_t *RegVal);
+
+#ifndef PHY_ADDR
+#define PHY_ADDR        ((uint32_t)CONFIG_ETH_STM32_HAL_PHY_ADDRESS)
+#endif
+
+#ifndef PHY_BSR
+#define PHY_BSR     ((uint16_t)0x0001U)
+#endif
+
+#ifndef PHY_LINKED_STATUS
+#define PHY_LINKED_STATUS       ((uint16_t)0x0004U)
+#endif
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+/**
+ * Definitions for ethernet driver of all others STM32 except STM32H7
+ */
+
+#define GET_TX_DESC_LIST_PTR(heth)   (&(((ETH_HandleTypeDef *)heth)-> \
+																TxDescList))
+
+#define GET_FIRST_DMA_TX_DESC(heth)  ((ETH_DMADescTypeDef *) \
+									  (((ETH_HandleTypeDef *)heth)-> \
```

Comment:
the last \ seems to be redundant as the next line is empty
indentation is also way off

Reply:
corrected